mcodingbot/plugins/python.py

```diff
@@ -0,0 +1,47 @@
+import re
+
+import crescent
+import hikari
+
+from mcodingbot.utils import Plugin
+
+plugin = Plugin()
+
+PEP_REGEX = re.compile(r"pep *(?P<pep>\d{1,4})", re.IGNORECASE)
+
+
+def get_pep_link(pep_number: int) -> str:
+    return f"<https://peps.python.org/pep-{pep_number:04}/>"
+
+
+@plugin.include
+@crescent.command(
+    name="pep", description="Find a Python Enhancement Proposal."
+)
+class PEPCommand:
+    pep_number = crescent.option(int, "the PEP number")
+
+    async def callback(self, ctx: crescent.Context) -> None:
+        await ctx.respond(get_pep_link(self.pep_number))
+
+
+@plugin.include
+@crescent.event
+async def on_message(event: hikari.MessageCreateEvent) -> None:
+    if not event.message.content or event.author.is_bot:
+        return
+
+    pep_refs = [
+        int(ref.groupdict()["pep"])
+        for ref in re.finditer(PEP_REGEX, event.message.content)
+    ]
+
+    if not pep_refs:
+        return
+
+    pep_links_message = "\n".join(
+        f"PEP {pep_number}: {get_pep_link(pep_number)}"
```

Comment:
for consistency, I would also format the pep_number to have zero-padding

Reply:
disagree, do you say pep 8 or pep 0008 :p

Reply:
I have to disagree with you. When people refer to the respective PEPs, they refer to them non-zero padded. It is perfectly fine as is.

Reply:
Yeah good point, nvm.